newIDE/app/src/MainFrame/index.js

```diff
@@ -2285,10 +2281,10 @@ const MainFrame = (props: Props) => {
         currentProject.setTemplateSlug(selectedExampleShortHeader.slug);
       if (source.projectName) currentProject.setName(source.projectName);
 
-      if (!destination) {
-        // If there is no destination, ensure the storageProvider is reset.
-        resetStorageProvider();
-      } else {
+      const destinationFileMetadata = destination
+        ? destination.fileMetadata
+        : null;
+      if (destination && destinationFileMetadata) {
```

Comment:
Note that the only thing that should be checked for calling `onSaveProjectAs` is for the existence of `onSaveProjectAs`. It should be called EVEN IF `destination.fileMetadata` is null.

Indeed, some storage provider, like DownloadFileStorageProvider (which allows you to download a zip of the game) don't have any fileMetadata but still support save as. So in case someone creates a project and would select this storage provider (of course it's a bit weird, but in theory it works), then it should work: onSaveProjectAs should work (and so the editor should open the example/project and then immediately call `onSaveProjectAs` to let you download the zip of the game).

Once we have this done, I think we're *correct*.